addons/mail/models/discuss/discuss_channel_member.py

```diff
@@ -106,8 +107,8 @@ def _compute_message_unread(self):
                       WHERE mail_message.model = 'discuss.channel'
                         AND mail_message.message_type NOT IN ('notification', 'user_notification')
                         AND (
-                            mail_message.id > discuss_channel_member.seen_message_id
-                         OR discuss_channel_member.seen_message_id IS NULL
+                            mail_message.id > discuss_channel_member.new_message_separator
+                         OR discuss_channel_member.new_message_separator IS NULL
```

Comment:
If we make this a required integer, default it to 0, and make sure to upgrade all DB, we might be able remove the NULL part?